pkg/goDB/engine/query.go

```diff
@@ -278,33 +297,72 @@ func (qr *QueryRunner) RunStatement(ctx context.Context, stmt *query.Statement)
 	return result, nil
 }
 
+func (qr *QueryRunner) runLiveQuery(ctx context.Context, mapChan chan hashmap.AggFlowMapWithMetadata, stmt *query.Statement) (wg *sync.WaitGroup) {
+	wg = new(sync.WaitGroup)
+
+	if !stmt.Live {
+		return
+	}
+
+	wg.Add(1)
+	go func() {
+		qr.captureManager.GetFlowMaps(ctx, goDB.QueryFilter(qr.query), mapChan, stmt.Ifaces...)
+		wg.Done()
+	}()
+
+	return
+}
+
 func createWorkManager(dbPath string, iface string, tfirst, tlast int64, query *goDB.Query, numProcessingUnits int) (workManager *goDB.DBWorkManager, nonempty bool, err error) {
-	workManager, err = goDB.NewDBWorkManager(dbPath, iface, numProcessingUnits)
+	workManager, err = goDB.NewDBWorkManager(query, dbPath, iface, numProcessingUnits)
 	if err != nil {
-		return nil, false, fmt.Errorf("could not initialize query work manager for interface '%s': %s", iface, err)
+		return nil, false, fmt.Errorf("could not initialize query work manager for interface '%s': %w", iface, err)
 	}
-	nonempty, err = workManager.CreateWorkerJobs(tfirst, tlast, query)
+	nonempty, err = workManager.CreateWorkerJobs(tfirst, tlast)
 	return
 }
 
-func parseIfaceList(dbPath string, ifacelist string) (ifaces []string, err error) {
+func parseIfaceList(dbPath string, ifacelist string) ([]string, error) {
 	if ifacelist == "" {
-		return nil, fmt.Errorf("no interface(s) specified")
+		return nil, errors.New("no interface(s) specified")
 	}
 
 	if strings.ToLower(ifacelist) == "any" {
-		ifaces, err = info.GetInterfaces(dbPath)
+		ifaces, err := info.GetInterfaces(dbPath)
 		if err != nil {
 			return nil, err
 		}
-	} else {
-		ifaces = strings.Split(ifacelist, ",")
-		for _, iface := range ifaces {
-			if iface == "" {
-				err = fmt.Errorf("interface list contains empty interface name")
-				return
-			}
+		return ifaces, nil
+	}
+
+	ifaces, err := validateIfaceNames(ifacelist)
+	if err != nil {
+		return nil, err
+	}
+
+	return ifaces, nil
+}
+
+var ifaceNameRegexp = regexp.MustCompile(`^[a-zA-Z0-9\.:_-]{1,15}$`)
```

Comment:
Thank you!

Reply:
Jup. Turns out it was unnecessary after all (see my explanation about the CodeQL issues), but still it's probably not the worst idea. @els0r Just to be sure: Does OSAG adhere to the default of max 15 character length for interface names or did someone override that kernel config setting by any chance? :sweat_smile: